File: mario.c
//Mario less, print half pyramid of specified height
#include <cs50.h>
#include <stdio.h>

//Function Declarations
void printf_rpchar(char chrct, int num_times);


int main(void)
{
    //Check user input is an integer [0,23]
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 0 || height > 23);

    //Print pyramid of specified height
    for (int i = 0 ; i < height; i ++)
    {
        printf_rpchar(' ', height - (i + 1));
        printf_rpchar('#', i + 2);
        printf("\n");
    }

}



//Function Definitions
void printf_rpchar(char chrct, int num_times)
{
    //Print a character multiple times.
    for (int j = 0; j < num_times; j++)
    {
        printf("%c", chrct);
    }
}

File: mario.py
"""Mario less experinced"""
from cs50 import get_int


def main():
    """Print some steps with a height specified by the user."""

    # Ask user for input
    while True:
        height = get_int("Height: ")

        # If height is a positive integer, 23 or less, break out of loop
        if (height >= 0) and (height <= 23):
            break

    # Intiate variables
    blocks = 2
    spaces = height - 1

    # Iterate over number of rows
    for row in range(height):

        # Print spaces
        print(' ' * spaces, end='')

        # Print blocks
        print('#' * blocks)

        # Increment varaibles
        blocks += 1
        spaces -= 1



# Implement main function
if __name__ == "__main__":
    main()